projects/packages/publicize/src/class-publicize.php

```diff
@@ -145,6 +145,12 @@ public function disconnect( $service_name, $connection_id, $_blog_id = false, $_
 	 * @return true
 	 */
 	public function receive_updated_publicize_connections( $publicize_connections ) {
+
+		// Clear the cache.
+		Connections::clear_cache();
+		// Populate the cache with the new data.
+		Connections::get_all();
```

Comment:
I think we might want to be careful with this. The reason for the check on `set_transient` below is because you can get race conditions where the request comes in while another API request is in flight. We could add a similar check where we set the transient, or remove the call to `get_all` and get the connections on the next request.

The problem with that is that it doesn't seem very resiliant if API calls fail, but that is the case currently, so it's probably an issue to tackle afterwards.

Reply:
We can simply remove that call there and let the call happen when we actually need the data. What do you think?